src/deterministic_secure_broadcast.rs

```diff
@@ -33,13 +33,16 @@ pub struct SecureBroadcastProc<A: SecureBroadcastAlgorithm> {
 
     // The set of members in this network.
     peers: BTreeSet<Actor>,
+
+    // Track number of invalid packets received from an actor
+    invalid_packets: BTreeMap<Actor, u64>,
```

Comment:
invalid_packets is just one type of fault.   perhaps/probably a more general fault tracking mechanism will be needed.